iothub_client/samples/pnp/pnp_temperature_controller/pnp_thermostat_component.c

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// Standard C header files
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+// PnP helper routines
+#include "pnp_protocol_helpers.h"
+#include "pnp_thermostat_component.h"
+
+// Core IoT SDK utilities
+#include "azure_c_shared_utility/xlogging.h"
+
+// The PnP convention defines the maximum length of a component 
+#define PNP_MAXIMUM_COMPONENT_LENGTH 64
```

Comment:
Yeah that's potentially useful moving to helpers.

Other components don't need this because they don't need to make copies into a malloc'd data structure so don't need to know max size up front.